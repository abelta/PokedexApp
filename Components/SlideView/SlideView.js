import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Animated } from 'react-native';

class SlideView extends PureComponent {
  constructor(props) {
    super(props);
    const { toRight, toLeft } = this.props;
    this.toRight = toRight && !toLeft;
    this.state = { left: new Animated.Value(this.toRight ? -100 : 100) };
  }

  componentDidMount() {
    this.slideIn();
  }

  slideIn() {
    const { left } = this.state;
    Animated.timing(left, { toValue: 0 }).start();
  }

  render() {
    const { children, style } = this.props;
    const { left } = this.state;
    return (
      <Animated.View
        style={[
          style,
          { left: left.interpolate({ inputRange: [0, 100], outputRange: ['0%', '100%'] }) },
        ]}
      >
        {children}
      </Animated.View>
    );
  }
}

SlideView.propTypes = {
  children: PropTypes.node.isRequired,
  style: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
  toLeft: PropTypes.bool,
  toRight: PropTypes.bool,
};

SlideView.defaultProps = {
  style: {},
  toLeft: false,
  toRight: true,
};

export default SlideView;
