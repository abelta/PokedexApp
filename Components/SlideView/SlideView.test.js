import React from 'react';
import renderer from 'react-test-renderer';
import SlideView from '.';

describe('SlideView', () => {
  it('renders properly', () => {
    const tree = renderer.create(<SlideView>Some text</SlideView>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
