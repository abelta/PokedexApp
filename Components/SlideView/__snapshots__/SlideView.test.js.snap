// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideView renders properly 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "left": "-100%",
    }
  }
>
  Some text
</View>
`;
